runtime/src/accounts.rs

```diff
@@ -309,7 +312,11 @@ impl Accounts {
         let requested_loaded_accounts_data_size_limit =
             Self::get_requested_loaded_accounts_data_size_limit(feature_set);
         let mut accumulated_accounts_data_size: usize = 0;
+        let mut application_fees: HashMap<Pubkey, u64> = HashMap::new();
+        let mut payer_index = 0;
 
+        // we reverse the iteration over keys so that we collect all the application fees
+        // and then validate fees at the last
```

Comment:
Not sure if understand this 100%. ~Will application_fee change transaction base fee (by `bank.calculate_fee()`) that used in `validate_fee_payer()`?~ OK, so the collected application_fees are added on top of tx's base_fee before `validate_fee_payer()`, then the payer_account will be deducted by `sum(application_fee) _ base_fee`. Am I correct? If so, wondering how would "bankless leader" validates fee payer without loading and exam-ing all accounts.

Reply:
Yes application fees are on top of base fees. So we need to load all the accounts to calculate application fees and then charge the payer application fees + base fees. Good question about the bankless leader though. I though the bankless leader will not execute transactions just order them. I think it will have to load accounts anyways to test if the transaction is valid and if the accounts do exist.

Reply:
gotcha. Yeah, need to think about bankless leader bit more. The idea was to cache each payer account's balance, which is populated/updated by replay stage. Leader adjust payer account's balance by transaction's base fee as it packs it into block, thus far without having to load accounts. With Application Fee, it seems necessary to load all accounts tho. Something to think about.

Reply:
Wondering if possible that leader can just read `application_fee` instruction from transaction to know how much the transaction is willing to pay application fee(s), much like what `compute_budget` does today? Leader only needs to know _how much_, doesn't need to care about accounts or anything else.

Reply:
Something like this : https://github.com/solana-foundation/solana-improvement-documents/pull/16#issuecomment-1416708523